lib/inspec/profile.rb

```diff
@@ -525,23 +525,29 @@ def info_from_parse
       # TODO - look at the various source contents
       # PASS 1: parse them using rubocop-ast
       #   Look for controls, top-level metadata, and inputs
-      # PASS 2: Using the control IDs, deterimine the extents - 
-      # line locations - of the coontrol IDs in each file, and 
+      # PASS 2: Using the control IDs, deterimine the extents -
+      # line locations - of the coontrol IDs in each file, and
       # then extract each source code block. Use this to populate the source code
       # locations and 'code' properties.
 
       # @source_reader.tests contains a hash mapping control filenames to control file contents
-      @source_reader.tests.each do |_control_filename, control_file_source|
+      @source_reader.tests.each_with_index do |(control_filename, control_file_source), index|
```

Comment:
This index is not what you want. It is not the index of the controls in the profile. It is just the number of the file. If There are three files in the profile, this index will range 0..2. 

You will want to pass the control filename as a parameter to Inspec::Profile::AstHelper::ControlIDCollector.new.

Reply:
I see your point!